src/main/java/com/woowacourse/matzip/domain/restaurant/RestaurantFindQueryFactory.java

```diff
@@ -0,0 +1,36 @@
+package com.woowacourse.matzip.domain.restaurant;
+
+import com.woowacourse.matzip.exception.InvalidSortConditionException;
+import java.util.Arrays;
+import lombok.Getter;
+
+@Getter
+public enum RestaurantFindQueryFactory {
+
+    ORDER_BY_RATING_DESC("RATING", "select r from Restaurant r left join Review rv on rv.restaurantId = r.id "
+            + "where (r.campusId = :campusId) and (:categoryId is null or r.categoryId = :categoryId) "
+            + "group by r.id order by avg(rv.rating) desc"),
+    ORDER_BY_NAME_ASC("SPELL", "select r from Restaurant r "
+            + "where (r.campusId = :campusId) and (:categoryId is null or r.categoryId = :categoryId) "
+            + "order by r.name"),
+    ORDER_BY_ID_DESC("DEFAULT", "select r from Restaurant r "
+            + "where (r.campusId = :campusId) and (:categoryId is null or r.categoryId = :categoryId) "
+            + "order by r.id desc"),
+    ;
+
+    private final String key;
+    private final String query;
+
+    RestaurantFindQueryFactory(String key, String query) {
+        this.key = key;
+        this.query = query;
+    }
+
+    public static String from(String sortCondition) {
+        return Arrays.stream(values())
+                .filter(condition -> condition.key.equalsIgnoreCase(sortCondition))
+                .findAny()
+                .orElseThrow(InvalidSortConditionException::new)
+                .query;
```

Comment:
```suggestion
        return Arrays.stream(values())
                .filter(condition -> condition.key.equalsIgnoreCase(sortCondition))
                .findAny()
                .map(RestaurantFindQueryFactory::getQuery)
                .orElseThrow(InvalidSortConditionException::new);
```
개인적으로 stream 문법시에 orElseThrow가 마지막에 나오고 결과에 대한 표현이 되어야한다고 생각해요. map이라는 문법이 존재하는 이유도 이와 마찬가지이구요.
그렇다면 query보다는 현재와 같은 형태나 `map(condition -> condition.key`는 어떨까하는데 어떻게 생각하시나요?